File: Rivington.IG.Web/Rivington.IG.API/src/app/modules/shared/models/ordermanagement/inspection-order/checklist/photos/phototype.ts
import { PhotoTypeGroup } from "./phototypegroup";
import { Enumeration } from "../../../../../enumeration";

export class PhotoType extends Enumeration {

    groupId? : string;
    phototypegroup? : PhotoTypeGroup;

    public constructor(init?:Partial<PhotoType>) {
        super();
        Object.assign(this, init);
    }
}

File: Rivington.IG.Web/Rivington.IG.Domain.Models/OrderManagement/InspectionOrderWildfireAssessment/PhotoType.cs
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Rivington.IG.Domain.Models.OrderManagement
{
    public class PhotoType : Enumeration
    {
        public string GroupId { get; set; }
        [ForeignKey(nameof(GroupId))]
        public PhotoTypeGroup Group { get; set; }

        public static void BuildModel(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<PhotoType>()
                .HasOne(a => a.Group).WithOne()
                .HasConstraintName($"FK_PhotoType_{nameof(PhotoTypeGroup)}");
        }
    }
}
